sa/storage-authority.go

```diff
@@ -242,13 +252,19 @@ func (ssa *SQLStorageAuthority) NewRegistration(reg core.Registration) (core.Reg
 	if err != nil {
 		return reg, err
 	}
-
-	err = tx.Insert(&reg)
+	rm, err := registrationToModel(&reg)
+	if err != nil {
+		return reg, err
+	}
+	err = tx.Insert(rm)
 	if err != nil {
 		tx.Rollback()
 		return reg, err
 	}
-
+	reg, err = modelToRegistration(rm)
+	if err != nil {
+		return reg, err
```

Comment:
Though, reading the code more again, I'm not sure why this needs to happen in a transaction.